src/utils/credentials.ts

```diff
@@ -1,38 +1,60 @@
 import fs from 'fs/promises';
 import type { DB, Credential } from '../types';
+import { encryptCredential, decryptCredential } from './crypto';
 
-export async function readCredentials(): Promise<Credential[]> {
-  const response = await fs.readFile('src/db.json', 'utf-8');
-  const db: DB = JSON.parse(response);
-  const credentials = db.credentials;
-  return credentials;
+export async function readCredentials(key: string): Promise<Credential[]> {
+  try {
+    const response = await fs.readFile('src/db.json', 'utf-8');
+    const db: DB = JSON.parse(response);
+    const credentials = db.credentials;
+    return credentials;
+    const decryptedCredentials = credentials.map((credential) => {
+      return (credential = decryptCredential(credential, key));
+    });
+    return decryptedCredentials;
+  } catch (error) {
+    throw new Error(`Error: ${error}`);
+  }
 }
 
-export async function getCredential(service: string): Promise<Credential> {
-  const credentials = await readCredentials();
+export async function getCredential(
+  service: string,
+  key: string
+): Promise<Credential> {
+  const credentials = await readCredentials(key);
   const credential = credentials.find(
     (item) => item.service.toLowerCase() === service.toLowerCase()
   );
 
   if (!credential) {
     throw new Error(`No credential found for service: ${service}`);
   } else {
-    return credential;
+    const decryptedCredential = decryptCredential(credential, key);
+    return decryptedCredential;
```

Comment:
The credential will already be decrypted at this point. So this will either return an empty string or do nothing. I would recommend removing the decryption from the readCredentials function as it will decrypt the whole database